src/ray/protobuf/usage.proto

```diff
@@ -0,0 +1,55 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+option cc_enable_arenas = true;
+
+package ray.usage;
+
+// This file defines your custom TagKey to record custom usage stats.
+//
+// Steps to record custom usage stats:
+// 1. Define your key in the TagKey enum (this means the key cannot be dynamic).
+// 2. Call `ray._private.usage.usage_lib.record_extra_usage_tag(k, v)` in your code.
+// 3. One of @pcmoritz, @thomasdesr should review and approve the data
+// collection. In particular, please make sure values are not fully dynamic
+// (no user-defined data like their file name, etc!)
+
+enum TagKey {
+  // Test only.
+  _TEST1 = 0;
+  _TEST2 = 1;
+
+  // RLlib
+  // The deep learning framework ("tf", "torch", etc.).
+  RLLIB_FRAMEWORK = 2;
+  // The algorithm name (only built-in algorithms).
+  RLLIB_ALGORITHM = 3;
+  // The number of workers as a string.
+  RLLIB_NUM_WORKERS = 4;
+
+  // Serve
+  // The public Python API version ("v1", "v2").
+  SERVE_API_VERSION = 5;
+  // The total number of running serve deployments as a string.
+  SERVE_NUM_DEPLOYMENTS = 6;
+  // The GCS storage type, which could be memory or redis.
+  GCS_STORAGE = 7;
+
+  // Core
+  // Total number of placement groups created.
+  PG_NUM_CREATED = 300;
```

Comment:
cc @pcmoritz can you review this?

Reply:
Yes, this looks good to me, thanks for the ping :)